aten/src/ATen/LegacyTHFunctionsCPU.cpp

```diff
@@ -332,6 +332,15 @@ Tensor _th_histc(const Tensor & self, int64_t bins, const Scalar& min, const Sca
 }
 
 std::tuple<Tensor &,Tensor &> _th_gels_out(const Tensor & self, const Tensor & A, Tensor & res1, Tensor & res2) {
+    TORCH_WARN_ONCE(
+      "torch.lstsq is deprecated in favor of torch.linalg.lstsq and will be removed in a future PyTorch release.\n",
+      "torch.linalg.lstsq has reversed arguments and does not return the QR decomposition in "
+      "the returned tuple, (it returns other information about the problem).\n",
+      "X, _ = torch.lstsq(B, A).solution[:A.size(1)]\n",
```

Comment:
This seems like a very specific invocation and manipulation of the result of torch.lstsq()

Maybe there's a more generic way to tell users to replace torch.lstsq() with torch.linalg.lstsq(), or maybe we need more time to deprecate torch.lstsq()?

Reply:
This also comes from a previous wrong design. `torch.lstsq` returns the solution and the residuals packed into the "solution" tensor.

Any user would have to unpack the solution and the residuals. That slicing there does the unpacking. I'll add this point to the warning.